### PR TITLE
Fixing initial DMs/GMs load

### PR DESCRIPTION
**Pull request by jespino**
#### Summary
Fixing initial DMs/GMs load

#### Ticket Link
[MM-17223](https://mattermost.atlassian.net/browse/MM-17223)

### CODE REVIEW

**Comment by grundleborg** [comment]
@jespino ci is failing

**Comment by jespino** [comment]
@grundleborg fixed

File: store/sqlstore/channel_store.go [hunk 1]
```diff
@@ -1939,7 +1939,7 @@ func (s SqlChannelStore) AnalyticsDeletedTypeCount(teamId string, channelType st
 
 func (s SqlChannelStore) GetMembersForUser(teamId string, userId string) (*model.ChannelMembers, *model.AppError) {
 	var dbMembers channelMemberWithSchemeRolesList
-	_, err := s.GetReplica().Select(&dbMembers, CHANNEL_MEMBERS_WITH_SCHEME_SELECT_QUERY+"WHERE ChannelMembers.UserId = :UserId AND Teams.Id = :TeamId", map[string]interface{}{"TeamId": teamId, "UserId": userId})
+	_, err := s.GetReplica().Select(&dbMembers, CHANNEL_MEMBERS_WITH_SCHEME_SELECT_QUERY+"WHERE ChannelMembers.UserId = :UserId AND (Teams.Id = :TeamId OR Teams.Id IS NULL)", map[string]interface{}{"TeamId": teamId, "UserId": userId})
```
Comment (enahum) [inline store/sqlstore/channel_store.go]: @jespino this does not WORK!!!!!!

it should be `Teams.Id= ''` as it is never `NULL`

the following Query works

```
SELECT
		ChannelMembers.*,
		TeamScheme.DefaultChannelGuestRole TeamSchemeDefaultGuestRole,
		TeamScheme.DefaultChannelUserRole TeamSchemeDefaultUserRole,
		TeamScheme.DefaultChannelAdminRole TeamSchemeDefaultAdminRole,
		ChannelScheme.DefaultChannelGuestRole ChannelSchemeDefaultGuestRole,
		ChannelScheme.DefaultChannelUserRole ChannelSchemeDefaultUserRole,
		ChannelScheme.DefaultChannelAdminRole ChannelSchemeDefaultAdminRole
	FROM
		ChannelMembers
	INNER JOIN
		Channels ON ChannelMembers.ChannelId = Channels.Id
	LEFT JOIN
		Schemes ChannelScheme ON Channels.SchemeId = ChannelScheme.Id
	LEFT JOIN
		Teams ON Channels.TeamId = Teams.Id
	LEFT JOIN
		Schemes TeamScheme ON Teams.SchemeId = TeamScheme.Id
WHERE ChannelMembers.UserId=:UserId AND (TeamId=:TeamId OR TeamId='');
```

Comment (enahum) [reply store/sqlstore/channel_store.go]: If you want to be extra sure then you can do `.. (TeamId=:TeamId OR TeamId='' OR TeamId IS NULL)`

Comment (enahum) [reply store/sqlstore/channel_store.go]: and definitely add a unit test for this!!!!

Comment (jespino) [reply store/sqlstore/channel_store.go]: I going to do this

**Pull request merged**